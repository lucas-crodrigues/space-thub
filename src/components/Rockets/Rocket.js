/* eslint-disable camelcase */
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';

const Rocket = (props) => {
  const {
    id, rocket_name, description, flickr_images,
  } = props;

  const dispatch = useDispatch();

  const handleReserve = () => {
    dispatch(console.log(`the rocket ${id}:${rocket_name} is reserved`));
  };

  return (
    <div className="rocket">
      <div className="rocketImage">
        <img src={flickr_images} alt={`${rocket_name}`} />
      </div>
      <div className="rocketInfo">
        <h3>{rocket_name}</h3>
        <p>{description}</p>
        <button type="button" onClick={handleReserve}>Reserve Rocket</button>
      </div>
    </div>
  );
};

Rocket.propTypes = {
  id: PropTypes.string.isRequired,
  rocket_name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  flickr_images: PropTypes.string.isRequired,
};

export default Rocket;
