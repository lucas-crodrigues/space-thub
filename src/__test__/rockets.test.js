import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import React from 'react';
import store from '../redux/configureStore';
import Rocket from '../components/Rockets/Rocket';
import UserRockets from '../components/user/userRockets';

describe('Test suite for all Rockets features', () => {
  test('check snapshot of rocket list', () => {
    const rocketInst = {
      id: 1,
      rocket_name: 'rocket 1',
      description: 'rockets description',
      reserved: false,
      flickr_images: ['img1', 'img2'],
    };

    const component = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Rocket
              rocket={rocketInst}
            />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(component).toMatchSnapshot();
  });

  test('check snapshot of rocket list 2', () => {
    const rocketInst2 = {
      id: 2,
      rocket_name: 'rocket 2',
      description: 'rockets description',
      reserved: false,
      flickr_images: ['img1', 'img2'],
    };

    const component = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Rocket
              rocket={rocketInst2}
            />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(component).toMatchSnapshot();
  });

  test('check snapshot of user rocket section', () => {
    const component = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <UserRockets name="rocket 1" wiki="wiki1" />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(component).toMatchSnapshot();
  });

  test('check snapshot of user rocket section', () => {
    const component = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <UserRockets name="rocket 2" wiki="wiki2" />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(component).toMatchSnapshot();
  });
});
