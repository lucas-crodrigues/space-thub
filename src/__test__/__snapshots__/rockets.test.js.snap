// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for all Rockets features check snapshot of rocket list 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rocket"
      >
        <div
          class="rocketImage"
        >
          <img
            alt="rocket 1"
            src="img1,img2"
          />
        </div>
        <div
          class="rocketInfo"
        >
          <h3>
            rocket 1
          </h3>
          <p>
            <small
              class="available"
            >
              Available
            </small>
            rockets description
          </p>
          <button
            class=""
            type="button"
          >
            Reserve rocket 1
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rocket"
    >
      <div
        class="rocketImage"
      >
        <img
          alt="rocket 1"
          src="img1,img2"
        />
      </div>
      <div
        class="rocketInfo"
      >
        <h3>
          rocket 1
        </h3>
        <p>
          <small
            class="available"
          >
            Available
          </small>
          rockets description
        </p>
        <button
          class=""
          type="button"
        >
          Reserve rocket 1
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test suite for all Rockets features check snapshot of rocket list 2 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rocket"
      >
        <div
          class="rocketImage"
        >
          <img
            alt="rocket 2"
            src="img1,img2"
          />
        </div>
        <div
          class="rocketInfo"
        >
          <h3>
            rocket 2
          </h3>
          <p>
            <small
              class="available"
            >
              Available
            </small>
            rockets description
          </p>
          <button
            class=""
            type="button"
          >
            Reserve rocket 2
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="rocket"
    >
      <div
        class="rocketImage"
      >
        <img
          alt="rocket 2"
          src="img1,img2"
        />
      </div>
      <div
        class="rocketInfo"
      >
        <h3>
          rocket 2
        </h3>
        <p>
          <small
            class="available"
          >
            Available
          </small>
          rockets description
        </p>
        <button
          class=""
          type="button"
        >
          Reserve rocket 2
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test suite for all Rockets features check snapshot of user rocket section 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="user-li"
      >
        <p>
          rocket 1
        </p>
        <a
          href="wiki1"
        >
          Learn More
        </a>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="user-li"
    >
      <p>
        rocket 1
      </p>
      <a
        href="wiki1"
      >
        Learn More
      </a>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test suite for all Rockets features check snapshot of user rocket section 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="user-li"
      >
        <p>
          rocket 2
        </p>
        <a
          href="wiki2"
        >
          Learn More
        </a>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="user-li"
    >
      <p>
        rocket 2
      </p>
      <a
        href="wiki2"
      >
        Learn More
      </a>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
